### PR TITLE
Adam C W5 Friday challenge

### PR DESCRIPTION
**Pull request by adam-conway**
This went well up until the spec changed from calling trips hikes. After that point a lot of confusion on how to move forward. Also had trouble getting my relationship tests to work.

### CODE REVIEW

File: spec/features/trips/visitor_visits_trips_show_from_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  scenario 'A visitor sees longest hiking distance' do
+    trip2 = Trip.create!(name: "Surfing", start_date: Date.strptime('8/29/2013', '%m/%d/%Y'), end_date: Date.strptime('9/10/2013', '%m/%d/%Y'))
+    trail1 = Trail.create!(length: 100, name: "best trail", address: "111 middle of forest")
+    trail2 = Trail.create!(length: 200, name: "ok trail", address: "222 middle of forest")
+    TripTrail.create!(trip_id: trip2.id, trail_id: trail1.id)
+    TripTrail.create!(trip_id: trip2.id, trail_id: trail2.id)
+
+    visit trip_path(trip2)
+
+    expect(page).to have_content("Longest hiking distance: 200")
```
Comment (AliSchlereth) [inline spec/features/trips/visitor_visits_trips_show_from_index_spec.rb]: I like how this assertion as a string that helps us know what the 200 means.

File: spec/models/trail_spec.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  #   it {is_expected.to have_many(:trip_trail)}
+  # end
+  describe 'Analysis methods' do
+    context 'Total hiking distance' do
+      it 'returns the total hiking distance' do
+        trail1 = Trail.create!(length: 100, name: "best trail", address: "111 middle of forest")
+        trail2 = Trail.create!(length: 200, name: "ok trail", address: "222 middle of forest")
+        trail3 = Trail.create!(length: 100, name: "bad trail", address: "333 middle of forest")
+
+        total = trail1.length + trail2.length + trail3.length
```
Comment (AliSchlereth) [inline spec/models/trail_spec.rb]: I love how you build total here, rather than simply asserting for 400!

File: spec/models/trail_spec.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+        expect(Trail.total_hiking_distance).to eq(total)
+      end
+    end
+    context 'average hiking distance' do
+      it 'returns the average hiking distance' do
+        trail1 = Trail.create!(length: 100, name: "best trail", address: "111 middle of forest")
+        trail2 = Trail.create!(length: 200, name: "ok trail", address: "222 middle of forest")
+
+        average = (trail1.length + trail2.length)/2
```
Comment (AliSchlereth) [inline spec/models/trail_spec.rb]: ✨

**Comment by adam-conway** [comment]
Appreciate the feedback!

**Pull request closed without merging**